### DISCUSSION TITLE
Expand/Collapse `SidebarItem` at runtime?

### DISCUSSION THREAD

**Discussion by sharpchen** [vuejs/vitepress#3308] [Q&A]
I'd like to auto expand the sidebarItem corresponds to the opened page when loading. Sidebar items are all collapsed by default.
```ts
    import { useSidebar } from 'vitepress/theme';
    import { onMounted } from 'vue';
    const { sidebarGroups } = useSidebar();
    onMounted(() => {
        // sidebarGroups.value[0].collapsed = false;
    });
```

**Comment by babakfp** [comment]
https://github.com/vuejs/vitepress/assets/44144724/efc767dc-c2ae-4fbe-9cc0-460e128bdfb3

Is this what you mean? This is by default. I have `collapsed: true,` for "Migrations".

Or do mean you want to expand an item when the user navigates to a specific page? So when the user is on page B, the sidebar item of page A to be open.

**Reply by sharpchen** [reply]
Yes exactly, even better if I could expand it when navigate. Any solution for this?

**Reply by babakfp** [reply]
I don't know to be honest. Maybe you can listen to URL changes and whenever you are in the correct path, you can add some classes/styles to the nav.

https://gomakethings.com/how-to-detect-when-the-browser-url-changes-with-vanilla-js/
